giftcertificate/dao/src/main/java/com/epam/esm/dao/impl/GiftCertificateJDBCTemplate.java

```diff
@@ -0,0 +1,227 @@
+    private static final String DELETE_GIFT_CERTIFICATE_HAS_TAG = "DELETE FROM `gift_db`.`gift_certificates_has_tags` WHERE `gift_certificates_id` = ?";
+    private static final String DELETE_GIFT_CERTIFICATE = "DELETE FROM gift_db.gift_certificates WHERE id = ?";
+    private static final String DEFAULT_SORT_ORDER = "asc";
+
+    /**
+     * Instance of JdbcTemplate for work with DB
+     */
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    /**
+     * Instance of GiftCertificateMapper for mapping data from resultSet
+     */
+    @Autowired
+    private GiftCertificateMapper giftCertificateMapper;
+
+    /**
+     * Instance of TagMapper for mapping data from resultSet
+     */
+    @Autowired
+    private TagMapper tagMapper;
+
+    /**
+     * Create GiftCertificate in DB
+     *
+     * @param giftCertificate we wont create in DB
+     * @return created GiftCertificate
+     * @throws DuplicateEntryDAOException if this GiftCertificate already exists in the DB
+     */
+    @Override
+    public GiftCertificate create(GiftCertificate giftCertificate) throws DuplicateEntryDAOException {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        giftCertificate.setCreateDate(LocalDateTime.now(ZoneId.systemDefault()));
+        giftCertificate.setLastUpdateDate(giftCertificate.getCreateDate());
+        try {
+            jdbcTemplate.update(connection -> {
+                PreparedStatement preparedStatement = connection.prepareStatement(INSERT_CERTIFICATE, Statement.RETURN_GENERATED_KEYS);
+                preparedStatement.setString(1, giftCertificate.getName());
+                preparedStatement.setString(2, giftCertificate.getDescription());
+                preparedStatement.setInt(3, giftCertificate.getPrice());
+                preparedStatement.setInt(4, giftCertificate.getDuration());
+                preparedStatement.setTimestamp(5, Timestamp.valueOf(giftCertificate.getCreateDate()));
+                preparedStatement.setTimestamp(6, Timestamp.valueOf(giftCertificate.getLastUpdateDate()));
+                return preparedStatement;
+            }, keyHolder);
+        } catch (Exception e) {
+            logger.error("Trying to create duplicate entry of GiftCertificate");
+            throw new DuplicateEntryDAOException(e.getMessage());
+
+        }
+        logger.info("GiftCertificate is created in DB");
+        giftCertificate.setId(Objects.requireNonNull(keyHolder.getKey()).intValue());
+
+        createGiftCertificateHasTag(giftCertificate);
+        return giftCertificate;
+    }
+
+    /**
+     * Read GiftCertificate from DB by id
+     *
+     * @param id long type parameter
+     * @return Optional<GiftCertificate>
+     * @throws IdNotExistDAOException if records with such id not exist in DB
+     */
+    @Override
+    public GiftCertificate read(long id) throws IdNotExistDAOException {
```

Comment:
I decided to not overcomplicate you life)) No need to read in one query, just again, move 2 calls in manager rather than having 2 dao calls here